### PR TITLE
Feat/render accept arguments

### PR DESCRIPTION
**Pull request by vilarfg**
related to : #44

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1281,8 +1281,10 @@ var lighterhtml = (function (document,exports) {
     };
   }; // generic content render
 
-  function render(node, callback) {
-    var value = update.call(this, node, callback);
+  function render() {
+    var node = arguments[0];
+    arguments.unshift(this);
```
Comment (WebReflection) [inline index.js]: arguments is not an array

File: index.js [hunk 2]
```diff
@@ -1367,10 +1369,13 @@ var lighterhtml = (function (document,exports) {
   }
 
   function update(reference, callback) {
+    var reference = arguments[0];
+    var callback = arguments[1];
+    arguments.splice(0, 2, this);
```
Comment (WebReflection) [inline index.js]: arguments is not an array

File: index.js [hunk 3]
```diff
@@ -1367,13 +1367,10 @@ var lighterhtml = (function (document,exports) {
...
 
   function update(reference, callback) {
-    var reference = arguments[0];
-    var callback = arguments[1];
-    arguments.splice(0, 2, this);
     var prev = current;
     current = wm.get(reference) || set(reference);
     current.i = 0;
-    var ret = callback.apply(arguments);
+    var ret = callback.apply(this, arguments.slice(2));
```
Comment (WebReflection) [inline index.js]: arguments is not an array

**Comment by WebReflection** [comment]
this is not tested/working, but I've also answered to your quest in the ticket

**Comment by vilarfg** [comment]
true: not tested nor working, I will think about what you said in the ticket and (maybe) rework this.

Thank you so much!

File: index.js [hunk 4]
```diff
@@ -1282,7 +1282,7 @@ var lighterhtml = (function (document,exports) {
   }; // generic content render
 
   function render(node) {
-    var value = update.apply(this, arguments);
+    var value = update.apply(this, Array.prototype.slice.call(arguments));
```
Comment (WebReflection) [inline index.js]: this is historically a perf killer:
https://www.jstips.co/en/javascript/avoid-modifying-or-passing-arguments-into-other-functions-it-kills-optimization/
https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments

I hope you starts getting why I'm not super keen to play around with arguments here

Comment (vilarfg) [reply index.js]: Oh, I was unaware of that
Thanks!

**Pull request closed without merging**